kong/db/dao/plugins.lua

```diff
@@ -4,7 +4,7 @@ local utils = require "kong.tools.utils"
 local Entity = require "kong.db.schema.entity"
 local DAO = require "kong.db.dao"
 local MetaSchema = require "kong.db.schema.metaschema"
-
+local Set = require "pl.Set"
```

Comment:
I would not be using the Penlight set object if I could prevent it. Since here you don't use the complex methods the object has, I'd simply write my own code here

Reply:
That was actually my first implementation. Then I thought "man, someone is going to look at my `create_set` function and ask why I haven't used one of the `utils` or penlight functions instead of reinventing the wheel" xD

Reply:
I'll go one step further: objects of type `"set"` in our schemas already ship with a metatable for set-style access. :)

However, I just double checked and their implementation is O(n). So I pushed a branch to fix this. You may want to cherry-pick this commit into your branch:

https://github.com/Kong/kong/commit/5e94c0ff0391c841855f7f1341775d9aaa991bd1

Reply:
Ok, since the schema sets already provide the functionality, I disposed of penlight and used them directly. I also cherry-picked the perf change.